src/lib/format.js

```diff
@@ -24,14 +24,18 @@ export function valueFromContext(ctx, id, args) {
 export function entityFromContext(ctx, id, args) {
   const entity = ctx.messages.get(id);
 
-  if (!entity)  {
+  if (entity === undefined)  {
     return [
       { value: id, attrs: null },
       [new L10nError(`Unknown entity: ${id}`)]
     ];
   }
 
-  const [value] = ctx.format(entity, args);
+  let value = null;
+
+  if (typeof entity === 'string' || Array.isArray(entity) || entity.val !== undefined) {
+    value = ctx.format(entity, args)[0];
+  }
```

Comment:
What if there's no value but one of the traits has the asterisk?

    key =
     *[html/title] Foo

I think it would be safest to just run this through the resolver and fix the resolver to know about `null` values?